apps/charterafrica/src/lib/ecosystem/airtable/processData.js

```diff
@@ -0,0 +1,194 @@
+import * as Sentry from "@sentry/nextjs";
+
+function getter(data, key) {
+  return key ? data?.[key] : null;
+}
+
+function mapSupportersToFields(supporters, config, tableData) {
+  const {
+    schema: { partnerTableColumns },
+  } = config;
+  const { name, url, logo } = partnerTableColumns;
+  const { partners } = tableData;
+  const mapped = supporters.map((id) => {
+    try {
+      const { fields } = partners.find((item) => id === item.id);
+      return {
+        name: getter(fields, name),
+        website: getter(fields, url),
+        logo: getter(fields, logo),
+      };
+    } catch (error) {
+      Sentry.captureMessage(error.message);
+      return null;
+    }
+  });
+  return mapped.filter(Boolean);
+}
+
+function mapSocialMediaToFields(socialMedia, config, tableData) {
+  const {
+    schema: { socialMediaTableColumns },
+  } = config;
+  const { name, url } = socialMediaTableColumns;
+  const { socialMedia: socialMediaData } = tableData;
+  const mapped = socialMedia.map((id) => {
+    try {
+      const { fields } = socialMediaData.find((item) => item.id === id);
+      return {
+        name: getter(fields, name),
+        website: getter(fields, url),
+      };
+    } catch (error) {
+      Sentry.captureMessage(error.message);
+      return null;
+    }
+  });
+  return mapped.filter(Boolean);
+}
+
+export function processTool(data, config, tableData) {
+  try {
+    const {
+      schema: { toolTableColumns },
+    } = config;
+    const theme = {
+      en: getter(data, toolTableColumns.theme.en)?.[0],
+      pt: getter(data, toolTableColumns.theme.pt)?.[0],
+      fr: getter(data, toolTableColumns.theme.fr)?.[0],
+    };
+    const description = {
+      en: getter(data, toolTableColumns.description.en),
+      pt: getter(data, toolTableColumns.description.pt),
+      fr: getter(data, toolTableColumns.description.fr),
+    };
```

Comment:
I'd have gone with something like this... Only process data that's going to be used.

```suggestion
    const { localized } = config;
    const locales = localized ? ['en', 'fr', 'pt'] : ['en'];
    const theme = locales.reduce((curr, acc) => {
      acc[curr] = getter(data, toolsTableColumns.theme[curr])?.[0];
      return acc;
    }, {};
    const description = locales.reduce((curr, acc) => {
      acc[curr] = getter(data, toolsTableColumns.description[curr]);
      return acc;
    }, {};
```